packages/web-ui/src/sections/Document/Sidebar/Files/index.tsx

```diff
@@ -1,98 +1,21 @@
 'use client'
 
-import {
-  forwardRef,
-  ReactNode,
-  useCallback,
-  useEffect,
-  useMemo,
-  useRef,
-  useState,
-} from 'react'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
 
-import { DropdownMenu, MenuOption } from '$ui/ds/atoms/DropdownMenu'
+import { MenuOption } from '$ui/ds/atoms/DropdownMenu'
 import { Icons } from '$ui/ds/atoms/Icons'
-import { Input } from '$ui/ds/atoms/Input'
-import Text from '$ui/ds/atoms/Text'
 import { cn } from '$ui/lib/utils'
-import {
-  FileTreeProvider,
-  useFileTreeContext,
-} from '$ui/sections/Document/Sidebar/Files/FilesProvider'
-import { useNodeValidator } from '$ui/sections/Document/Sidebar/Files/useNodeValidator'
-import { useTempNodes } from '$ui/sections/Document/Sidebar/Files/useTempNodes'
 
+import { FileTreeProvider, useFileTreeContext } from './FilesProvider'
+import NodeHeaderWrapper, {
+  ICON_CLASS,
+  type IndentType,
+} from './NodeHeaderWrapper'
+import { useNodeValidator } from './useNodeValidator'
 import { useOpenPaths } from './useOpenPaths'
+import { useTempNodes } from './useTempNodes'
 import { Node, SidebarDocument, useTree } from './useTree'
 
-const ICON_CLASS = 'min-w-6 h-6 text-muted-foreground'
-
-type IndentType = { isLast: boolean }
-
-function IndentationBar({
-  indentation,
-  hasChildren,
-}: {
-  hasChildren: boolean
-  indentation: IndentType[]
-}) {
-  return indentation.map((indent, index) => {
-    const anyNextIndentIsNotLast = !!indentation
-      .slice(index)
-      .find((i) => !i.isLast)
-    const showBorder = anyNextIndentIsNotLast ? false : indent.isLast
-    return (
-      <div key={index} className='h-6 min-w-6'>
-        {index > 0 ? (
-          <div className='-ml-[3px] relative w-6 h-full flex justify-center'>
-            {hasChildren || !showBorder ? (
-              <div className='bg-border w-px h-8 -mt-1' />
-            ) : (
-              <div className='relative -mt-1'>
-                <div className='border-l h-2.5' />
-                <div className='absolute top-2.5 border-l border-b h-2 w-2 rounded-bl-sm' />
-              </div>
-            )}
-          </div>
-        ) : null}
-      </div>
-    )
-  })
-}
-
-type WrapperProps = {
-  open: boolean
-  node: Node
-  selected?: boolean
-  onClick?: () => void
-  children: ReactNode
-  indentation: IndentType[]
-}
-const NodeHeaderWrapper = forwardRef<HTMLDivElement, WrapperProps>(function Foo(
-  { node, open, selected = false, onClick, children, indentation },
-  ref,
-) {
-  return (
-    <div
-      ref={ref}
-      onClick={onClick}
-      className={cn(
-        'max-w-full group/row flex flex-row my-0.5 cursor-pointer',
-        {
-          'hover:bg-muted': !selected,
-          'bg-accent': selected,
-        },
-      )}
-    >
-      <IndentationBar
-        indentation={indentation}
-        hasChildren={open && node.children.length > 0}
-      />
-      {children}
-    </div>
-  )
-})
```

Comment:
Now ☝️ a document and a folder share all the presentation layers and they only pass the necessary props. This way design is unified. This was a leftover refactor I had pending from initial file tree implementation